File: crawl/database/job.py
from datetime import datetime
from database.entity import Entity
from utils.tools import builder

@builder
class Job(Entity):
    securityId: str   
    lid: str          
    jobName: str
    jobType: str
    companyId: str
    url: str
    salary: str
    salaryFloor: int
    salaryCeiling: int
    crawlDate: datetime
    city: str
    region: str
    experience: str
    degree: str
    industry: str
    title: str
    skills: str
    description: str
    table_name:str = "job"
    key_name : str= "lid"
    
    def __post_init__(self):
        self.region = self.region.strip()
        self.degree = self.degree.strip()
        self.city = self.city.strip()
        
    

File: src/main/java/com/jobagent/entity/Job.java
package com.jobagent.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Job {
    private String securityId;
    private String id;
    private String jobName;
    private String jobType;
    private String companyId;
    private String companyName;
    private String url;
    private String salary;
    private int salaryFloor;
    private int salaryCeiling;
    private String crawlDate;
    private String city;
    private String region;
    private String experience;
    private String degree;
    private String industry;
    private String title;
    private String skills;
    private String description;
    private String scale;
    private String stage;
    private String welfare;
    private Boolean viewed;
    private Boolean sentCv;
}

